android/src/org/kiwix/kiwixmobile/KiwixMobileActivity.java

```diff
@@ -1748,7 +1738,13 @@ public void restoreTabStates() {
   private void manageExternalLaunchAndRestoringViewState(Bundle savedInstanceState) {
 
     if (getIntent().getData() != null) {
-      String filePath = getIntent().getData().getPath();
+      String filePath = FileUtils.getLocalFilePathByUri(getApplicationContext(), getIntent().getData());
+
+      if (filePath == null) {
+        Toast.makeText(KiwixMobileActivity.this, "Unable to load file", Toast.LENGTH_LONG).show();
```

Comment:
Can you localise this please. You can use the existing string error_filenotfound